### PR TITLE
Adding libpsm3 package

### PR DESCRIPTION
**Pull request by dodecatheon**
The libpsm3 package encapsulates the steps required for installing and running with PSM3 libraries from Intel.

### CODE REVIEW

File: var/spack/repos/builtin/packages/libpsm3/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libpsm3/package.py]: Please use the new syntax/decorator for the maintainer (see https://spack.readthedocs.io/en/latest/packaging_guide.html#maintainers).

File: var/spack/repos/builtin/packages/libpsm3/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libpsm3/package.py]: You can still use the `self.enable_or_disable()` helper here with the `variant` option (see https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html#the-variant-keyword).

File: var/spack/repos/builtin/packages/libpsm3/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libpsm3/package.py]: Same suggestion about leveraging a helper method.

File: var/spack/repos/builtin/packages/libpsm3/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libpsm3/package.py]: I recommending alphabetizing these by variant name for maintainability.

File: var/spack/repos/builtin/packages/libpsm3/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libpsm3/package.py]: Why not use `self.enable_or_disable("atomics")` here?

(https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html#enable-or-disable)

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed the version sha256. Should use the newer maintainers syntax. Recommend alphabetizing the variants and increasing leverage of helper methods.

File: var/spack/repos/builtin/packages/libpsm3/package.py
Comment (dodecatheon) [reply var/spack/repos/builtin/packages/libpsm3/package.py]: Done, see latest commit

File: var/spack/repos/builtin/packages/libpsm3/package.py
Comment (dodecatheon) [reply var/spack/repos/builtin/packages/libpsm3/package.py]: Done, see latest commit

File: var/spack/repos/builtin/packages/libpsm3/package.py
Comment (dodecatheon) [reply var/spack/repos/builtin/packages/libpsm3/package.py]: Done, see latest commit

File: var/spack/repos/builtin/packages/libpsm3/package.py
Comment (dodecatheon) [reply var/spack/repos/builtin/packages/libpsm3/package.py]: Done, see latest commit

File: var/spack/repos/builtin/packages/libpsm3/package.py
Comment (dodecatheon) [reply var/spack/repos/builtin/packages/libpsm3/package.py]: Done, see latest commit

**Comment by dodecatheon** [comment]
@tldahlgren , made required changes -- do you have time to take another look? All that is necessary before merging is rebase/merging with develop.

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**
